tests/functions/test_gemini.py

```diff
@@ -129,3 +131,81 @@ def test_generate_videos(self, reset_db: None) -> None:
             assert video_stream['height'] == 720, metadata
             assert video_stream['duration_seconds'] == duration, metadata
             assert audio_stream['duration_seconds'] == duration, metadata
+
+    @pytest.mark.parametrize('default_params', [True, False], ids=['default_params', 'custom_params'])
+    def test_generate_embeddings(self, reset_db: None, default_params: bool) -> None:
+        skip_test_if_not_installed('google.genai')
+        skip_test_if_no_client('gemini')
+        from pixeltable.functions.gemini import generate_embedding
+
+        t = pxt.create_table('test', {'rowid': pxt.Int, 'text': pxt.String})
+
+        # Test embeddings as a computed column
+        config = None
+        expected_dim = 1536
+        if not default_params:
+            expected_dim = 3072
+            config = {'output_dimensionality': expected_dim}
+        t.add_computed_column(embedding=generate_embedding(t.text, config=config))
+        assert t.embedding.col.col_type.matches(ts.ArrayType((expected_dim,), np.dtype('float32'))), (
+            t.embedding.col.col_type
+        )
+        validate_update_status(
+            t.insert(
+                [
+                    {'rowid': 1, 'text': 'Pixeltable is a great tool for AI workload orchestration and storage'},
+                    {'rowid': 2, 'text': 'The quick brown fox jumps over the lazy dog.'},
+                    {'rowid': 3, 'text': 'The five boxing wizards jump quickly.'},
+                ]
+            ),
+            expected_rows=3,
+        )
+        for row in t.collect():
+            embedding = row['embedding']
+            assert isinstance(embedding, np.ndarray)
+            assert embedding.dtype == np.float32
+            assert embedding.shape == (expected_dim,)
+
+        # Test embeddings as an index
+        if default_params:
+            t.add_embedding_index(t.text, embedding=generate_embedding)
```

Comment:
Won't this fail?

[Edit: I see; we're specifying a default model, but I'd argue we shouldn't be; see above comment]